tests/test_jetpack.py

```diff
@@ -5,53 +5,60 @@
 import nose
 
 from js_helper import _do_real_test_raw as _js_test
-from validator.testcases.markup.markuptester import MarkupParser
-import validator.testcases.jetpack as jetpack
+from tests.test_frameworks import (MockXPI, accepted_version,
+                                   assert_framework_files, disposable_metadata)
+from validator import constants
 from validator.errorbundler import ErrorBundle
+from validator.testcases import content, libraries
+from validator.testcases.markup.markuptester import MarkupParser
 from validator.xpi import XPIManager
 
 
-def _do_test(xpi_package, allow_old_sdk=True):
+# These files are a subset of the framework tests which are specific
+# to Jetpack. They're currently split into a separate file since
+# they're considerably more comprehensive than any other framework
+# tests, and mostly predate the current generalized tests.
 
-    err = ErrorBundle()
-    jetpack.inspect_jetpack(err, xpi_package, allow_old_sdk=allow_old_sdk)
-    return err
+with open("tests/resources/bootstrap.js") as bootstrap_file:
+    bootstrap = bootstrap_file.read()
+with open("tests/resources/bootstrap-jpm.js") as bootstrap_file:
+    bootstrap_jpm = bootstrap_file.read()
 
 
-class MockXPI(object):
+def _do_test(xpi_package, allow_old_sdk=True):
 
-    def __init__(self, resources):
-        self.resources = resources
+    err = ErrorBundle()
+    content.test_packed_packages(err, xpi_package)
 
-    def read(self, name):
-        if isinstance(self.resources[name], bool):
-            return ""
-        return self.resources[name]
+    if allow_old_sdk:
+        ignore_ids = [
+            ("testcases_frameworks", "validate_jetpack_cfx",
+             "outdated_version")]
 
-    def __iter__(self):
-        for name in self.resources.keys():
-            yield name
+        err.warnings = [e for e in err.warnings
+                        if e["id"] not in ignore_ids]
 
-    def __contains__(self, name):
-        return name in self.resources
+    print err.print_summary(verbose=True)
+    return err
 
 
 def test_not_jetpack():
     """Test that add-ons which do not match the Jetpack pattern are ignored."""
 
     err = _do_test(MockXPI({"foo": True, "bar": True}))
-    assert not err.errors
-    assert not err.warnings
-    assert not err.notices
+    eq_(err.errors, [])
```

Comment:
why replace `assert` with `eq_` in this case? Maybe use `assert err.errors == []` if you want a precise comparison?